data/script.js

```diff
@@ -0,0 +1,73 @@
+// Update interval set to same value in Configuration.h
+const AIR_QUALITY_SENSOR_UPDATE_SECONDS = ^SENSOR-UPDATE-SECONDS^;
+const INTERVAL_TAGS = ["current", "tenMinutes", "hour", "day"];
+
+// DOM node mappings for metric value and color
+const valueDivs = INTERVAL_TAGS.reduce(
+  (acc, tag) => ({ ...acc, [tag]: document.getElementById(`value-${tag}`) }),
+  {},
+);
+const bgDivs = INTERVAL_TAGS.reduce(
+  (acc, tag) => ({ ...acc, [tag]: document.getElementById(`bg-${tag}`) }),
+  {},
+);
+
+function showAQI(evt, avr_window) {
+  var i, tabcontent, tablinks;
+  tabcontent = document.getElementsByClassName("aqi-display");
+  for (i = 0; i < tabcontent.length; i++) {
+    tabcontent[i].style.display = "none";
+  }
+  tablinks = document.getElementsByClassName("aqi_window_option");
+  for (i = 0; i < tablinks.length; i++) {
+    tablinks[i].className = tablinks[i].className.replace(" active", "");
+  }
+  document.getElementById(avr_window).style.display = "block";
+  evt.currentTarget.className += " active";
+}
+
+function getAQIStatusColor(aqiValue) {
+  if (aqiValue <= 50) {
+    return 'aqi-green';
+  } else if (aqiValue <= 100) {
+    return 'aqi-yellow';
+  } else if (aqiValue <= 150) {
+    return 'aqi-orange';
+  } else if (aqiValue <= 200) {
+    return 'aqi-red';
+  } else if (aqiValue <= 300) {
+    return 'aqi-purple';
+  } else {
+    return 'aqi-maroon';
+  }
+}
```

Comment:
This is what the original `COLOR-*` tokens were intended to cover, as they would get replaced by the `aui-*` CSS class names in the C++ `Application::processRootPageHTML()` method. By also implementing this mappingsg in JS, we now have two implementations of the same mapping, which could create its own class of maintenance problems over time. I would prefer we stick to the token placement approach, which keeps the function in the C++ as the logic is also needed to set the LED color (which must happen in the C++).

Reply:
Removed the frontend logic. The controller's `/json` endpoint now returns the appropriate CSS class name with each metric value.